classes/Product.php

```diff
@@ -6088,18 +6089,22 @@ public function getWsPositionInCategory()
      */
     public function setWsPositionInCategory($position)
     {
-        if ($position < 0) {
-            WebserviceRequest::getInstance()->setError(500, $this->trans('You cannot set a negative position, the minimum for a position is 0.', array(), 'Admin.Catalog.Notification'), 134);
+        if ($position <= 0) {
+            WebserviceRequest::getInstance()->setError(500, $this->trans('You cannot set a negative position or 0, the minimum for a position is 1.', array(), 'Admin.Catalog.Notification'), 134);
+            return false;
         }
         $result = Db::getInstance()->executeS('
             SELECT `id_product`
             FROM `' . _DB_PREFIX_ . 'category_product`
             WHERE `id_category` = ' . (int) $this->id_category_default . '
             ORDER BY `position`
         ');
-        if (($position > 0) && ($position + 1 > count($result))) {
-            WebserviceRequest::getInstance()->setError(500, $this->trans('You cannot set a position greater than the total number of products in the category, minus 1 (position numbering starts at 0).', array(), 'Admin.Catalog.Notification'), 135);
+        if (($position > 0) && ($position > count($result))) {
+            WebserviceRequest::getInstance()->setError(500, $this->trans('You cannot set a position greater than the total number of products in the category, (position numbering starts at 1).', array(), 'Admin.Catalog.Notification'), 135);
```

Comment:
What about _You cannot set a position greater than the total number of products in the category, starting at 1._ instead?